Sell/Product/ProductImplementing.swift

```diff
@@ -0,0 +1,30 @@
+//
+//  ProductImplementing.swift
+//  RealifeTech
+//
+//  Created by Mickey Lee on 17/06/2021.
+//  Copyright © 2021 Realife Tech. All rights reserved.
+//
+
+import Foundation
+
+public class ProductImplementing {
```

Comment:
I don't see the need for this layer. The only thing the product feature does is allow us to use product repository methods, so I think we should expose ProductProvidable/ProductRepository under Sell.

I can see you would want a ProductImplementing interface if we plan on grouping _other_ product features together in `Sell.Product`, but if we do that, we're going to end up with the same kind of interface duplication & feature buildup in the Product module as we were worried about in the Sell module. When we discussed this on Wednesday, I thought we would just have

**Sell**
- `Product: ProductProvidable` (Product repo)
- `FulfilmentPoint: FulfilmentPointProvidable` (fp repo)
- Other future feature, maybe it uses product, but we expose it directly on this level.

- [ ] Let me now your thoughts

Reply:
The responsibility of the Sell module is only networking at the moment. I am fine to reduce the implementation layer and keep the repository layer only. However, I thought having two layers is to resolve the problem listed on the [Confluence](https://livestyled.atlassian.net/wiki/spaces/Tech/pages/3033399316/ID-1098+-+Apollo+GraphQL+Repository+Design#Part-1%3A-We-make-repositories(workers)-which-implement-an-individual-set-of-routes)
> When we add other features to Sell which aren’t just Apollo based operations, they will be mixed in and it will be hard to decouple and reason about.

I guess the interface might still be `Sell.product.makeProductDetailView()` or so in the future, so we can add the view factory to the implementation layer directly. However, it's uncertain and I am happy to only expose the repository layer and leave it when there are more methods added to the Sell's components.

Reply:
There may be a confusion here. The issue with Sell described in that document was one where Sell directly contained repository logic. If we move that logic into a repository, and expose the repository in Sell, then all the features are fully decoupled and sell is just an interface for accessing them.

As for `Sell.product.makeProductDetailView()` I think we should wait until we have more features to implement. We may want to group things under a Product interface, we may want to separate things into Sell.Data and Sell.View components, we may want to keep all the features in Sell like Sell.productData (exposes repo) Sell.productView (exposes view factory). It doesn't seem like there's really a right answer at this point.